src/main/java/core/basesyntax/SalaryInfo.java

```
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private LocalDate parsedDateFrom;
+    private LocalDate parsedDateTo;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder resultInfo = new StringBuilder();
+
+        // reading and parsing Strings dateFrom, dateTo
+        try {
+            parsedDateFrom = LocalDate.parse(dateFrom, dateFormatter);
+            parsedDateTo = LocalDate.parse(dateTo, dateFormatter);
+            resultInfo.append("Report for period ");
+            resultInfo.append(dateFrom).append(" - ").append(dateTo)
+                    .append(System.lineSeparator());
+        } catch (NullPointerException | DateTimeException e) {
+            System.out.println("Date should be not NULL! and be in \"dd.MM.yyyy\" format");
+            System.out.println(e.getMessage());
+        }
```

The `NullPointerException` might occur if `dateFrom` or `dateTo` is null. The task description does not specify how to handle this case, so you can either throw the exception or handle it gracefully. If you choose to handle it, make sure to log the error message and return a default value. However, throwing the exception might be more appropriate as it indicates that the input is invalid.